driver/src/main/java/org/neo4j/driver/internal/net/SocketConnection.java

```diff
@@ -69,15 +72,34 @@ public SocketConnection( BoltServerAddress address, SecurityPlan securityPlan, L
             this.responseHandler = new SocketResponseHandler();
         }
 
-        this.socket = new SocketClient( address, securityPlan, logger );
-        socket.start();
+        this.socket.start();
+    }
+
+    // for mocked socket testing
+    SocketConnection( SocketClient socket, Logger logger )
+    {
+        this.socket = socket;
+        this.logger = logger;
+
+        if( logger.isDebugEnabled() )
+        {
+            this.responseHandler = new LoggingResponseHandler( logger );
+        }
+        else
+        {
+            this.responseHandler = new SocketResponseHandler();
+        }
+
+        this.socket.start();
     }
 
     @Override
     public void init( String clientName, Map<String,Value> authToken )
     {
+        Collector.InitCollector initCollector = new Collector.InitCollector();
         queueMessage( new InitMessage( clientName, authToken ), initCollector );
         sync();
+        this.serverInfo = new InternalServerInfo( socket.address(), initCollector.server() );
```

Comment:
`InternalServerInfo` constructor's second arg is called `String version` but here we pass `initCollector.server()` for it. This is a bit confusing. Maybe `InitCollector#server()` can be renamed to `#serverVersion()`?